jolpica_api/ergastapi/serializers.py

```diff
@@ -118,6 +106,44 @@ def get_qualifying(self, race: Round):
     def get_sprint(self, race: Round):
         return self.get_session_date_time(race, SessionType.SPRINT_RACE)
 
+    def get_sprint_qualifying(self, race: Round):
+        return self.get_session_date_time(race, SessionType.SPRINT_QUALIFYING1)
+
+    class Meta:
+        model = Round
+        fields = ["season", "round", "url", "raceName", "Circuit", "date", "time"]
+
+
+class ListRaceSerializer(serializers.ListSerializer):
+    def to_representation(self, data: list[Round]) -> dict:
+        races = list()
+        for round_ in data:
+            # check if there is any kind of sprint sesssion and if yes, use the sprint weekend format for that year
+            if round_.sessions.filter(type__startswith="S").exists():
```

Comment:
This line is the reason for the large amount of queries.

In the view we prefetch the sessions, however this line doesn't use that cache as it performs further filtering (and so has to fetch from the DB again).
To resolve this we can do some filtering locally in python

Reply:
Yes, I figured out that this line is the problem. I just didn't figure out how to solve it.
So does ``round_.sessions.all()`` not need to make another database query? I wasn't expecting this to reduce the amount of queries. If ``.filter()`` is a query, I expected that ``.all()`` would need to be a query as well.

Reply:
Yeah, the prefetch related does a query for ".all()" by default, to prefetch a filter some more complex custom Prefetching is required.
It is all a bit magic, and not the most clear as the filtered Prefetching actually creates new fields on the queryset, i think there is some examples for other views such as the driver view